api/src/DataProvider/Extension/TopBookPaginationExtension.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataProvider\Extension;
+
+use ApiPlatform\Core\DataProvider\ArrayPaginator;
+use ApiPlatform\Core\DataProvider\Pagination;
+use App\Entity\TopBook;
+
+final class TopBookPaginationExtension implements TopBookCollectionExtensionInterface
+{
+    private array $collection;
+    private array $context;
+    private Pagination $pagination;
+
+    public function __construct(Pagination $pagination)
+    {
+        $this->pagination = $pagination;
+    }
+
+    /**
+     * Returns the paginator object.
+     */
+    public function getResult(array $collection, string $resourceClass, string $operationName = null, array $context = []): iterable
+    {
+        $this->collection = $collection;
+        $this->context = $context;
+
+        return new ArrayPaginator($this->collection, $this->getOffset(), $this->getItemsPerPage());
```

Comment:
The idea was to return the Pagination object here, which includes all the pagination details

Reply:
Fixed.